prerequisites/build-functions/set_or_print_url.sh

```diff
@@ -10,6 +10,12 @@ set_or_print_url()
   # Get package name from argument passed with  -p, -D, -P, -V, or -U
   package_to_build="${arg_p:-${arg_D:-${arg_P:-${arg_U:-${arg_V:-${arg_B}}}}}}"
 
+if [[ ! -z ${arg_u:-}  ]]; then
+  # User specified a URL from which to download the package
+  url_tail=${arg_u##*/} # set url_tail to text_after_final_slash
+  url_head=${arg_u%%${url_tail}} # set url_head text before url_tail
```

Comment:
I don't think we need greedy filtering here, a single `%` will be safer. Also both RHS should be quoted for safety